kerboscript_tests/functest31.ks

```diff
@@ -0,0 +1,38 @@
+//
+// I am not too sure if this is a feature we want to advertise yet,
+// but it does seem to be doable:
+//
+print "Testing varying args logic".
+
+// A function that accepts varying args.
+// the first arg is a number: how many more args to expect.
+// the rest of the args are just printed for testing.
+function print_var_args {
+  parameter how_many.
+  
+  // read parameters, filling a local string.
+  local my_string is "".
+  local i is 1.
+  until i > how_many {
+    parameter next_arg.
+    set my_string to my_string + " " + next_arg.
+    set i to i + 1.
+  }
```

Comment:
The problem mentioned here was fixed by switching from doing a POP to check the argbottom, to doing a PEEK to check the argbottom, and leaving it up to OpcodeReturn later to pop the argbottom just like it did before this PR.  Thus a premature return will still pop the argbottom even if it happens before the actual argbottom check that occurs after the lastmost parameter.